cdap-ui/app/directives/log-viewer/log-viewer.html

```diff
@@ -14,38 +14,136 @@
+    <div class="timeline-buffer">
+    </div>
+
+  </div>
+
+  <!-- LOGS TABLE -->
+  <div class="logs-table">
+    <table class="table table-bordered">
+      <thead>
+        <th class="time"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.isTimeDisplayed">
+          Time
+        </th>
+
+        <th class="level"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.isLevelDisplayed">
+          <span uib-dropdown>
+            <span uib-dropdown-toggle>Level <span class="fa fa-caret-down"></span></span>
+            <ul uib-dropdown-menu ng-click="$event.stopPropagation();">
+              <li ng-repeat="option in LogViewer.logEvents">
+                <input type="checkbox" ng-click="LogViewer.includeEvent(option)"><span ui-sref='test'>{{option}}</span></li>
+              </li>
+            </ul>
+          </span>
+        </th>
+
+        <th class="source"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.isSourceDisplayed">
+          Source
+        </th>
+
+        <th class="message"
+            ng-if="LogViewer.isMessageDisplayed"
+            ng-class="{'expanded': LogViewer.isMessageExpanded }">
+          <span class="fa"
+                ng-class="{ 'fa-arrow-left': !LogViewer.isMessageExpanded,
+                  'fa-arrow-right': LogViewer.isMessageExpanded}"
+                ng-click="LogViewer.isMessageExpanded = !LogViewer.isMessageExpanded"></span>
+          <span>Message</span>
+        </th>
+      </thead>
+
+      <tbody>
+        <tr ng-repeat="log in LogViewer.data | filter:LogViewer.eventFilter | filter:searchText">
+          <!-- TIME -->
+          <td class="time"
+              ng-if="!LogViewer.isMessageExpanded && LogViewer.isTimeDisplayed">
+            <span>{{ log.time }}</span>
+          </td>
+
+          <!-- LEVEL -->
+          <td class="level"
+              ng-if="!LogViewer.isMessageExpanded && LogViewer.isLevelDisplayed">
+            <span>{{ log.level }}</span>
+          </td>
+
+          <!-- SOURCE -->
+          <td class="source"
+              ng-if="!LogViewer.isMessageExpanded && LogViewer.isSourceDisplayed">
+            <span>{{ log.source }}</span>
+          </td>
+
+          <!-- MESSAGE -->
+          <td class="message"
+              ng-if="LogViewer.isMessageDisplayed"
+              ng-class="{'stack-trace-expanded': log.isStackTraceExpanded,
+                'stack-trace-available': log.message.stackTrace }">
+            <div class="message-main clearfix">
+              <div class="icon-column"
+                   ng-if="log.message.stackTrace"
+                   ng-click="log.isStackTraceExpanded = !log.isStackTraceExpanded">
+                <span class="fa"
+                      ng-class="{'fa-plus': !log.isStackTraceExpanded,
+                        'fa-minus': log.isStackTraceExpanded }">
+                </span>
+              </div>
+              <div class="message-column">
+                {{ log.message.content }}
```

Comment:
We could probably have single binding here. The message content is not going to change inside the `ng-repeat` right?